src/tests/ftest/dfuse/root_container.py

```diff
@@ -138,25 +139,33 @@ def verify_create_delete_containers(self, pool, device, cont_count, hosts, mount
 
         expected = pool_space_before - cont_count * tmp_file_count * tmp_file_size
         self.log_step(
-            "Verify the pool free space <= {expected} after creating {cont_count} containers")
-        pool_space_after = pool.get_pool_free_space(device)
-        self.log.info("Pool space <= Expected")
-        self.log.info("%s <= %s", pool_space_after, expected)
-        self.assertTrue(pool_space_after <= expected)
+            f"Verify the pool free space <= {expected} after creating {cont_count} containers")
+        pool_space_after = self._get_pool_free_space(pool, device, expected)
+        if pool_space_after > expected:
+            self.fail(f"Pool free space exceeds {expected} after creating {cont_count} containers")
 
         self.log_step(f"Destroy half of the {cont_count} new sub containers ({cont_count // 2})")
         for _ in range(cont_count // 2):
             containers[-1].destroy(1)
             containers.pop()
 
+        # Wait for pool free space to reach expected value or timeout
         expected = pool_space_after + ((cont_count // 2) * tmp_file_count * tmp_file_size)
+        max_loops = 10
+        loop = 0
         self.log_step(
-            "Verify the pool free space >= {expected} after destroying half of the containers")
-        pool_space_after_cont_destroy = pool.get_pool_free_space(device)
-        self.log.info("After container destroy")
-        self.log.info("Free Pool space >= Expected")
-        self.log.info("%s >= %s", pool_space_after_cont_destroy, expected)
-        self.assertTrue(pool_space_after_cont_destroy >= expected)
+            f"Verify the pool free space >= {expected} after destroying half of the containers")
+        while loop < max_loops:
+            loop += 1
+            self.log.debug(
+                "Check if the pool free space >= %s (loop %s/%s)", expected, loop, max_loops)
+            current = self._get_pool_free_space(pool, device, expected)
+            if current >= expected:
+                break
+            if loop >= max_loops:
+                self.fail(
+                    f"Pool free space less than {expected} after destroying half of the containers")
+            time.sleep(int(60 / max_loops))
```

Comment:
FYI `sleep` does work with floats
```
>>> import time
>>> time.sleep(1.12345)
>>>
```

And also FYI if you just want to make sure it's a whole number you can use `//`. But the final type depends on the types passed in, so not always an int
```
>>> 123 / 4
30.75 # float
>>> 123 // 4
30 # int
>>> int(123 / 4)
30 # int
>>> 123.0 // 4
30.0 # float!
```